src/components/landing/UpcomingAnnouncements.tsx

```diff
@@ -0,0 +1,117 @@
+"use client";
+
+import { useState, useEffect } from "react";
+import HourglassIcon from "@/icons/HourglassIcon";
+import AnnouncementBanner from "../shared/AnnouncementBanner";
+import { Announcement } from "@/models/Announcement";
+
+interface Props {
+  announcements: Announcement[];
+}
+
+export default function UpcomingAnnouncements({ announcements }: Props) {
+  const futureAnnouncements = announcements
+    .filter((announcement) => new Date(announcement.startDate) > new Date())
+    .sort(
+      (a, b) =>
+        new Date(a.startDate).getTime() - new Date(b.startDate).getTime()
+    );
+  const [timeRemaining, setTimeRemaining] = useState({
+    days: 0,
+    hours: 0,
+    minutes: 0,
+    seconds: 0,
+  });
+
+  const calculateTimeRemaining = (startDate: Date) => {
+    const now = new Date();
+    const timeDiff = startDate.getTime() - now.getTime();
+    if (timeDiff <= 0) return { days: 0, hours: 0, minutes: 0, seconds: 0 };
+
+    const seconds = Math.floor(timeDiff / 1000);
+    const minutes = Math.floor(seconds / 60);
+    const hours = Math.floor(minutes / 60);
+    const days = Math.floor(hours / 24);
+
+    return {
+      days: Math.min(days, 99),
+      hours: hours % 24,
+      minutes: minutes % 60,
+      seconds: seconds % 60,
+    };
+  };
```

Comment:
I would factor this to a helper function under `utils.ts`. Write specifications and test cases.